internal/hardware/hardware_test.go

```diff
@@ -0,0 +1,53 @@
+package hardware
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/jakub-dzon/k4e-operator/models"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	logger       logr.Logger = log.FromContext(context.Background())
+	testHostname string      = "the-best-host"
+)
+
+func TestNilInput(t *testing.T) {
```

Comment:
Please use Ginkgo for writing tests and use `// given - // when - // then convention` to clearly show what is the setup, what is the actually tested call and what is the verification . See https://github.com/jakub-dzon/k4e-operator/tree/main/internal/utils.